### PR TITLE
Makes light fixtures actually glow in the dark while on (And adds the necessary groundwork for other above lighting plane stuff)

### PR DESCRIPTION
**Pull request by deathride58**
Title.

Before:
dreamseeker_2018-02-21_16-44-11

After:
dreamseeker_2018-02-21_16-33-29

:cl: deathride58
add: Lights will now actually glow in the dark!
/:cl:

### CODE REVIEW

**Comment by DaedalusGame** [comment]
this is unironically a good change

**Comment by Xhuis** [comment]
*That* is cool. I can think of so many things that can use this!

**Comment by ChangelingRain** [comment]
downside: anything using this will render above things that aren't using it, even if it shouldn't render above, such as megafauna/alien queens/other shit

**Comment by DaedalusGame** [comment]
are you sure

**Comment by ChangelingRain** [comment]
yes I've tried stuff like this before

**Comment by XDTM** [comment]
Doesn't this mean they'll ignore colored lighting? Might look jarring in some cases.
Also, it seems that this is only really useful because of night dimming, since light emitters are already in lighted turfs anyway

**Comment by DaedalusGame** [comment]
also technically it's possible to prevent this by adding a fullblack overlay at abovelightinglayer to... everything

**Comment by deathride58** [comment]
APCs, air alarms, fire alarms, airlocks, and various other machinery around the station could probably make use of the above lighting layer, as they're often in dimmed areas where their glowy bits end up getting darkened by the ambient lighting.

Light fixtures, and pretty much every other piece of machinery that could make use of the above lighting layer, are all almost always above most other objects. This means the oddities that occur when one object is above another object that has an above lighting overlay will hopefully be way too rare in actual gameplay to justify the memory impact of adding full-black overlays to every single object in the game.

**Comment by epochayur** [comment]
at some point, I'd love to see this on anything with a backlit display, like APCs, request consoles, morgue lights, etc.

**Comment by XDTM** [comment]
don't forget light colors; this means that even if there's a red floodlight in the room you'll see even the weakest lightbulbs in their normal color

**Comment by deathride58** [comment]
The issue with light overlays not being influenced by the room's lighting has been somewhat resolved via light overlays using the light's power for their alpha. Unfortunately, this will only apply during night shifts and on Citadel, due to the sparse usage of light sources with a light power below 1 in this codebase. I could probably decrease the min/max alpha values for light fixture overlays, though.

For the issue with light fixtures with non-standard colors suddenly having overlays that aren't influenced nearly as much by the light fixture's actual color, I could resolve this by making light overlays inherit their source's light color, but that'll have odd results for subtle, non-incandescent-like colors. Only way to resolve the issue with subtle colors looking odd with the light overlays would be to make light overlays greyscale, but that'll introduce the problem of lights losing their stylization, and will make light fixture overlay colors physically inaccurate in some cases, both of which are pretty big bags of worms that I'd rather not drown in.

**Comment by WJohn** [comment]
Could you do this with computers, while you're at it?

**Review by WJohn — DISMISSED** [review]

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/mutable_appearance.dm [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
 	..()
-	plane = FLOAT_PLANE // No clue why this is 0 by default yet images are on FLOAT_PLANE
+	plane = maplane // No clue why this is 0 by default yet images are on FLOAT_PLANE
 						// And yes this does have to be in the constructor, BYOND ignores it if you set it as a normal var
 
 // Helper similar to image()
-/proc/mutable_appearance(icon, icon_state = "", layer = FLOAT_LAYER)
-	var/mutable_appearance/MA = new()
+/proc/mutable_appearance(icon, icon_state = "", layer = FLOAT_LAYER, plane = FLOAT_PLANE)
+	var/mutable_appearance/MA = new(plane)
```
[the remark this answers is not in the document]
Comment (Cyberboss) [reply code/datums/mutable_appearance.dm]: Didn't know that, forget me

**Review by Cyberboss — DISMISSED** [review]

File: code/datums/mutable_appearance.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (deathride58) [reply code/datums/mutable_appearance.dm]: Just tested, seems like the issue where mutable appearances can't have their plane set outside of a proc within the mutable appearance itself has been fixed!

**Review by Cyberboss — CHANGES_REQUESTED** [review]
[It's not gonna work out](https://github.com/tgstation/tgstation/search?utf8=%E2%9C%93&q=mutable_appearance+ma+%3D+new&type=). I'm certain those aren't all the cases either

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**